### PR TITLE
nautilus: mgr/volumes: Add a per subvolume trash

### PR DESCRIPTION
**Pull request by ShyamsundarR**
Backport tracker: https://tracker.ceph.com/issues/47157

---

backport of #36773
parent tracker: https://tracker.ceph.com/issues/47154

---

Also includes commit(s) from #36744 to enable conflict free backport of the parent PR.

### CODE REVIEW

File: doc/cephfs/fs-volumes.rst [hunk 1]
```diff
@@ -91,12 +91,8 @@ List subvolume groups using::
 
     $ ceph fs subvolumegroup ls <vol_name>
 
-Create a snapshot (see :doc:`/cephfs/experimental-features`) of a
-subvolume group using::
-
-    $ ceph fs subvolumegroup snapshot create <vol_name> <group_name> <snap_name>
-
-This implicitly snapshots all the subvolumes under the subvolume group.
+.. note:: Subvolume group snapshot feature is no longer supported in mainline CephFS (existing group
```
Comment (ShyamsundarR) [inline doc/cephfs/fs-volumes.rst:94]: Possibly need this one word change here,

```suggestion
.. note:: Subvolume group snapshot feature is no longer supported in nautilus CephFS (existing group
```

Comment (batrick) [reply doc/cephfs/fs-volumes.rst:94]: Go ahead and push a follow-up commit with that change.

**Comment by batrick** [comment]
https://pulpito.ceph.com/pdonnell-2020-08-27_01:15:36-fs-wip-pdonnell-testing-20200826.234047-nautilus-distro-basic-smithi/

File: src/pybind/mgr/volumes/fs/operations/versions/subvolume_base.py [hunk 2]
```diff
@@ -159,24 +198,26 @@ def set_attrs(self, path, size, isolate_namespace, pool, uid, gid):
                 raise VolumeException(-e.args[0], e.args[1])
 
         # set uid/gid
+        uid = attrs.get("uid")
         if uid is None:
             uid = self.group.uid
         else:
             try:
-                uid = int(uid)
```
Comment (ajarr) [inline src/pybind/mgr/volumes/fs/operations/versions/subvolume_base.py:166]: This  has introduced the regression,
https://pulpito.ceph.com/pdonnell-2020-08-27_01:15:36-fs-wip-pdonnell-testing-20200826.234047-nautilus-distro-basic-smithi/5379256/

File: src/pybind/mgr/volumes/fs/operations/versions/subvolume_base.py [hunk 3]
```diff
@@ -159,24 +198,26 @@ def set_attrs(self, path, size, isolate_namespace, pool, uid, gid):
...
                     raise ValueError
             except ValueError:
                 raise VolumeException(-errno.EINVAL, "invalid UID")
+
+        gid = attrs.get("gid")
         if gid is None:
             gid = self.group.gid
         else:
             try:
-                gid = int(gid)
```
Comment (ajarr) [inline src/pybind/mgr/volumes/fs/operations/versions/subvolume_base.py:175]: This has introduced the regression,
https://pulpito.ceph.com/pdonnell-2020-08-27_01:15:36-fs-wip-pdonnell-testing-20200826.234047-nautilus-distro-basic-smithi/5379256/

**Comment by ajarr** [comment]
See,
https://github.com/ceph/ceph/pull/36851

I've added the uid, gid fix here. And one of the commit messages said it cherry-picked from 2 commits, so fixed that too.

**Comment by ShyamsundarR** [comment]
Build started: https://shaman.ceph.com/builds/ceph/wip-shyamsundarr-nautilus-2020-08-27/ (includes https://github.com/ceph/ceph/pull/36404 and is on latest nautilus, and hence already includes #36804)

**Comment by ShyamsundarR** [comment]
jenkins test api

**Comment by batrick** [comment]
This just needs rebased as #36804 is merged.

**Comment by ShyamsundarR** [comment]
> This just needs rebased as #36804 is merged.

Rebased.

Also all tests passed: http://pulpito.front.sepia.ceph.com/shyamsundarr-2020-08-27_15:38:13-fs-wip-shyamsundarr-nautilus-2020-08-27-distro-basic-smithi/

**Pull request merged**